source/opt/optimizer.cpp

```diff
@@ -75,12 +75,13 @@ bool Optimizer::Run(const uint32_t* original_binary,
   if (module == nullptr) return false;
 
   auto status = impl_->pass_manager.Run(module.get());
-  if (status == opt::Pass::Status::SuccessWithChange) {
+  const bool succeeded = status != opt::Pass::Status::Failure;
+  if (succeeded) {
```

Comment:
Actually we can do a bit better.  We don't have to generate the binary if it's pass without change *and* the target binary storage aliases the input binary.